traildb/traildb.py

```diff
@@ -414,6 +415,42 @@ def create_filter(self, event_filter):
         return TrailDBEventFilter(self, event_filter)
 
 class TrailDBEventFilter(object):
+    """
+    Converts a query defined in terms of Python collections to a
+    `tdb_event_filter` which can be passed to various TrailDB functions.
+    Performs some validation when parsing the query.
+
+    Queries are boolean expressions defined from terms and clauses.  A term is
+    defined using a tuple:
+
+    (field_name, "value") -- match records with field_name == "value"
+    (field_name, "value", False) -- match records with field_name == "value"
+    (field_name, "value", True) -- match records with field_name != "value"
```

Comment:
Changing that would break the existing API -- I just documented the API that already exists.  My understanding is that we absolutely don't want to make backwards incompatible changes unless we have to, at least for the C API.  Does that apply to the Python API as well?